src/SOS/Strike/strike.cpp

```diff
@@ -9174,6 +9141,79 @@ GetClrMethodInstance(
     ___in ULONG64 NativeOffset,
     ___out IXCLRDataMethodInstance** Method);
 
+typedef std::tuple<DacpMethodDescData, DacpCodeHeaderData, HRESULT> ExtractionCodeHeaderResult;
+
+ExtractionCodeHeaderResult extractCodeHeaderData(DWORD_PTR methodDesc, DWORD_PTR dwStartAddr);
+HRESULT displayGcInfo(BOOL fWithGCInfo, const DacpCodeHeaderData& codeHeaderData);
+HRESULT displayIntermediateLanguage(BOOL bIL, const DacpCodeHeaderData& codeHeaderData,
+                                    std::unique_ptr<CLRDATA_IL_ADDRESS_MAP[]>& map,
+                                    ULONG32& mapCount);
+
+GetILAddressResult GetILAddress(const DacpMethodDescData& MethodDescData)
+{
+    GetILAddressResult error = std::make_tuple(NULL, nullptr);
+    TADDR ilAddr = NULL;
+    struct DacpProfilerILData ilData;
+    ReleaseHolder<ISOSDacInterface7> sos7;
+    if (SUCCEEDED(g_sos->QueryInterface(__uuidof(ISOSDacInterface7), &sos7)) &&
+        SUCCEEDED(sos7->GetProfilerModifiedILInformation(MethodDescData.MethodDescPtr, &ilData)))
+    {
+        if (ilData.type == DacpProfilerILData::ILModified)
+        {
+            ExtOut("Found profiler modified IL\n");
+            ilAddr = TO_TADDR(ilData.il);
+        }
+    }
+
+    // Factor this so that it returns a map from IL offset to the textual representation of the decoding
+    // to be consumed by !u -il
+    // The disassemble function can give a MethodDescData as well as the set of keys IL offsets
+
+    // This is not a dynamic method, print the IL for it.
+    // Get the module
+    DacpModuleData dmd;    
+    if (dmd.Request(g_sos, MethodDescData.ModulePtr) != S_OK)
+    {
+        ExtOut("Unable to get module\n");
+        return error;
+    }
+
+    ToRelease<IMetaDataImport> pImport(MDImportForModule(&dmd));
+    if (pImport == NULL)
+    {
+        ExtOut("bad import\n");
+        return error;
+    }
+
+    if (ilAddr == NULL)
+    { 
+        ULONG pRva;
+        DWORD dwFlags;
+        if (pImport->GetRVA(MethodDescData.MDToken, &pRva, &dwFlags) != S_OK)
+        {
+            ExtOut("error in import\n");
+            return error;
+        }    
+
+        CLRDATA_ADDRESS ilAddrClr;
+        if (g_sos->GetILForModule(MethodDescData.ModulePtr, pRva, &ilAddrClr) != S_OK)
+        {
+            ExtOut("FindIL failed\n");
+            return error;
+        }
+
+        ilAddr = TO_TADDR(ilAddrClr);
+    }
+
+    if (ilAddr == NULL)
+    {
+        ExtOut("Unkown error in reading function IL\n");
```

Comment:
Feel free to correct typo as you found it - for example, this should be `Unknown` instead of `Unkown`.